bitnami/zookeeper/templates/statefulset.yaml

```diff
@@ -268,13 +272,17 @@ spec:
           {{- if .Values.readinessProbe.enabled }}
           readinessProbe:
             exec:
-              {{- if not .Values.service.tls.disable_base_client_port }}
-              command: ['/bin/bash', '-c', 'echo "ruok" | timeout {{ .Values.readinessProbe.probeCommandTimeout }} nc -w {{ .Values.readinessProbe.probeCommandTimeout }} localhost {{ .Values.service.port }} | grep imok']
+              {{- if .Values.readinessProbe.useAdmin }}
+              command: ['/bin/bash', '-c', 'curl -s -m {{ .Values.readinessProbe.probeCommandTimeout }} http://localhost:8080/commands/ruok | grep error | grep null']
```

Comment:
Expected output from this check when the server can host client traffic:
```
$ curl -s http://localhost:8080/commands/ruok
{
  "command" : "ruok",
  "error" : null
}
```

Output from this check with the server cannot host client traffic but the jvm is still running:
```
$ curl -s http://localhost:8080/commands/ruok
{
  "command" : "ruok",
  "error" : "This ZooKeeper instance is not currently serving requests"
}
```

Reply:
Please do not hardcode the port to 8080. The port where Zookeeper is listening for connections is dynamic (see the block below in the environment section), thus we cannot hardcode it here.
```yaml
            - name: ZOO_PORT_NUMBER
              value: {{ .Values.service.port | quote }}
```

Reply:
8080 is the default value for `zookeeper.admin.serverPort`. Nothing in the chart or the underlaying Docker container config allows this port to be directly set (outside of ad hoc use of the 'jvmFlags' value).

'ZOO_PORT_NUMBER' corresponds to the `clientPort` value set in the conf file supplied to the ZooKeeper instance.